indico/modules/events/sessions/controllers/display.py

```diff
@@ -0,0 +1,49 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2016 European Organization for Nuclear Research (CERN).
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 3 of the
+# License, or (at your option) any later version.
+#
+# Indico is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Indico; if not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import unicode_literals
+
+from flask import session
+
+from indico.core.db.sqlalchemy.principals import PrincipalType
+from indico.modules.events.sessions.models.principals import SessionPrincipal
+from indico.modules.events.sessions.models.sessions import Session
+from indico.modules.events.sessions.views import WPDisplayMySessionsConference
+from MaKaC.webinterface.rh.base import RHProtected
+from MaKaC.webinterface.rh.conferenceDisplay import RHConferenceBaseDisplay
+
+
+class RHDisplaySessionList(RHConferenceBaseDisplay, RHProtected):
+    def _checkProtection(self):
+        RHProtected._checkProtection(self)
+        RHConferenceBaseDisplay._checkProtection(self)
+
+    def _process(self):
+        sessions = (self.event_new.sessions
+                    .filter(Session.acl_entries.any(SessionPrincipal.has_management_role('coordinate')),
+                            ~Session.is_deleted)
+                    .all())
+        user_sessions = set()
```

Comment:
can be simplified a lot:

    user_sessions = {sess for sess in sessions if any(user in entry.principal for entry in iter_acl(sess.acl_entries))}